docs/reference/security/external-principals-aws.md

```diff
@@ -0,0 +1,128 @@
+
+{% include toc.html %}
+
+## Overview 
+
+lakeFS supports authenticating users programmatically using AWS IAM roles instead of using static lakeFS access and secret keys.
+The method enables you to bound IAM principal ARNs to lakeFS users.
+A single lakeFS user may have many AWS's principle ARNs attached to it. When a client is authenticating to a lakeFS server with an AWS's session, the actions performed by the client are on behalf of the user attached to the ARN.
+
+### Using Session Names
+
+The bound ARN can be attached to a single lakeFS user with or without SessionName, serving different users.
+For example, consider the following mapping: 
+
+| Principal ARN                                       | lakeFS User |
+|-----------------------------------------------------|-------------|
+| arn:aws:sts::123456:assumed-role/Dev                | foo         |
+| arn:aws:sts::123456:assumed-role/Dev/john@acme.com  | john        |
+
+if the bound ARN were `arn:aws:sts::123456:assumed-role/Dev/<SessionName>` it would allow any principal assuming `Dev` role in AWS account `123456` to login to it.
+If the `SessionName` is `john@acme.com` then lakeFS would return token for `john` user
+
+### How AWS authentication works
+
+The AWS STS API includes a method, `sts:GetCallerIdentity`, which allows you to validate the identity of a client. The client signs a GetCallerIdentity query using the AWS Signature v4 algorithm and sends it to the lakeFS server. 
+
+The `GetCallerIdentity` query consists of four pieces of information: the request URL, the request body, the request headers and the request method. The AWS signature is computed over those fields. The lakeFS server reconstructs the query using this information and forwards it on to the AWS STS service. Depending on the response from the STS service, the server authenticates the client.
+
+Notably, clients don't need network-level access themselves to talk to the AWS STS API endpoint; they merely need access to the credentials to sign the request. However, it means that the lakeFS server does need network-level access to send requests to the STS endpoint.
+
+Each signed AWS request includes the current timestamp to mitigate the risk of replay attacks. In addition, lakeFS allows you to require an additional header, `X-LakeFS-Server-ID` (added by default), to be present to mitigate against different types of replay attacks (such as a signed `GetCallerIdentity` request stolen from a dev lakeFS instance and used to authenticate to a prod lakeFS instance). 
+
+It's also important to note that Amazon does NOT appear to include any sort of authorization around calls to GetCallerIdentity. For example, if you have an IAM policy on your credential that requires all access to be MFA authenticated, non-MFA authenticated credentials will still be able to authenticate to lakeFS using this method.
```

Comment:
Nice
Maybe we should add here [this](https://blog.xargs.io/post/2023-07-01-use-presigned-aws-sts-get-caller-identity-for-authentication/) article for reasoning?